src/compile/data/calculate.ts

```diff
@@ -53,24 +51,64 @@ export class CalculateNode extends DataFlowNode {
     return parent;
   }
 
+  /**
+   * Merge calculate nodes. This method either integrates the bin config from the other node
+   * or if this node already has a bin config, renames the corresponding signal in the model.
+   */
+  public merge(other: CalculateNode) {
+    for (const key of keys(other.calculates)) {
+      if (key in this.calculates) {
+        // make sure we are not merging something incompatible
+        if (this.calculates[key].calculate !== other.calculates[key].calculate) {
+          /* istanbul ignore next: This should never happen. */
```

Comment:
What if you have two layers that calculate two different things "as" the same variable?  

If it shouldn't happen, why add it in the code?